### PR TITLE
added documentation for mod_auth/models.py

### PR DESCRIPTION
**Pull request by pranavsid98**
Documentation for mod_auth/models.py

### CODE REVIEW

File: mod_auth/models.py [hunk 1]
```diff
@@ -23,35 +23,70 @@ class User(Base):
     role = Column(Role.db_type())
 
     def __init__(self, name, role=Role.user, email=None, password=''):
+	'''
```
Comment (canihavesomecoffee) [inline mod_auth/models.py]: Please use double quotes instead of single ones (ref. PEP257)

File: mod_auth/models.py [hunk 2]
```diff
@@ -23,35 +23,70 @@ class User(Base):
...
+	Parametrized constructor for the User model
```
Comment (canihavesomecoffee) [inline mod_auth/models.py]: I'm missing the arguments of the constructor.

File: mod_auth/models.py [hunk 3]
```diff
@@ -23,35 +23,70 @@ class User(Base):
...
+	Represent a User Model by its 'name' Field.
+	'''
         return '<User %r>' % self.name
 
     @staticmethod
     def generate_hash(password):
+	'''
+	Generates a Hash value for a password
+	:param password: The password to be hashed
+	:type password: str
```
Comment (canihavesomecoffee) [inline mod_auth/models.py]: :return and :rtype are missing. They provide some help when using an IDE. Please add for all methods that return something.

**Comment by pranavsid98** [comment]
Requested changes made.

**Comment by pranavsid98** [comment]
Requested changes made

File: mod_auth/models.py [hunk 4]
```diff
@@ -23,35 +23,98 @@ class User(Base):
     role = Column(Role.db_type())
 
     def __init__(self, name, role=Role.user, email=None, password=''):
+	"""
+	Parametrized constructor for the User model
+
+	:param name: The value of the 'name' field of User model
+	:type name: str
+	:param role: The value of the 'role' field of User model
+	:type role: Role Model
```
Comment (canihavesomecoffee) [inline mod_auth/models.py]: Almost. Mention Role should be enough.

File: mod_auth/models.py [hunk 5]
```diff
@@ -23,35 +23,98 @@ class User(Base):
...
+	:return : Compares role of the passed Model with 'Admin' and returns result	
+	:rtype: boolean
+	"""
         return self.role == Role.admin
 
     def has_role(self, name):
+	"""
+	Checks whether the User has a particular role
+
+	:param name: Name of the user
```
Comment (canihavesomecoffee) [inline mod_auth/models.py]: Role to check, not he name of the user

File: mod_auth/models.py [hunk 6]
```diff
@@ -23,35 +23,98 @@ class User(Base):
...
+	:type name: str
+	:return : Checks whether a passed User has a role.
```
Comment (canihavesomecoffee) [inline mod_auth/models.py]: Not fully ;) We have a user object, and call it like user.has_role('admin'), so this phrase isn't really correct.

File: mod_auth/models.py [hunk 7]
```diff
@@ -23,35 +23,98 @@ class User(Base):
...
+	:type new_password: str
+	"""
         self.password = self.generate_hash(new_password)
 
     @property
     def is_admin(self):
+	"""
+	Verifies if a User is the admin
+
+	:return : Compares role of the passed Model with 'Admin' and returns result
```
Comment (canihavesomecoffee) [inline mod_auth/models.py]: We don't pass anything. We just check if this user has an admin role or not.

**Comment by pranavsid98** [comment]
Requested changes made

**Pull request merged**